website/content/docs/configuration/seal/pkcs11.mdx

```diff
@@ -0,0 +1,156 @@
+---
+sidebar_label: PKCS#11 Unseal
+description: |-
+  The PKCS11 seal configures OpenBao to use an HSM with PKCS11 as the
+  autounseal mechanism.
+---
+# `pkcs11` seal
+
+The PKCS11 seal configures OpenBao to use an HSM with PKCS11 as the autounseal
+mechanism. OpenBao's HSM PKCS11 support is activated by one of the following:
+
+- The presence of a `seal "pkcs11"` block in OpenBao's configuration file
+- The presence of the environment variable `BAO_HSM_LIB` set to the library's
+  path as well as `BAO_SEAL_TYPE` set to `pkcs11`. If enabling via environment
+  variable, all other required values (i.e. `BAO_HSM_SLOT`) must be also
+  supplied.
+
+:::info
+
+Unlike Vault Enterprise, OpenBao requires key material to be created externally
+before initializing the instance. OpenBao will not attempt to create its own
+key material.
+
+:::
+
+## Requirements
+
+The following software packages are required for OpenBao:
+
+- PKCS#11 compatible HSM integration library. Depending on any given HSM, some
+  functions (such as key generation) may have to be performed manually.
+- The cgo-enabled OpenBao binary for your architecture, complete with relevant
+  dynamic libraries such as `glibc`.
+
+## `pkcs11` example
+
+This example shows configuring HSM PKCS11 seal through the OpenBao configuration
+file by providing all the required values:
+
+```hcl
+seal "pkcs11" {
+  lib = "/usr/lib64/softhsm/libsofthsm.so"
+  token_label = "OpenBao"
+  pin = "4321"
+  key_label = "bao-root-key-rsa"
+  rsa_oaep_hash = "sha1"
+}
+```
+
+To create the key in the HSM, use a utility like `pkcs11-tool`:
+
+```bash
+$ pkcs11-tool --module "/usr/lib64/softhsm/libsofthsm.so" --token-label "OpenBao" --so-pin 1234 --pin 4321 --keypairgen --key-type rsa:4096 --label "bao-root-key-rsa"
+```
+
+## `pkcs11` parameters
+
+These parameters apply to the `seal` stanza in the OpenBao configuration file:
+
+- `lib` `(string: <required>)`: The path to the PKCS#11 library shared object
+  file. May also be specified by the `BAO_HSM_LIB` environment variable.
+
+  ~> **Note:** Depending on your HSM, the value of the `lib` parameter may be
```

Comment:
Needs to be formatted properly

Reply:
@JanMa @DanGhita This is fixed and should be ready for review. I don't know how easy it is to build a containerized test of this so I might suggest we do manual testing for now.